pkg/generator/appconfiguration/generator/workload/magic_env_var.go

```diff
@@ -0,0 +1,133 @@
+package workload
+
+import (
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	SecretEnvParser    MagicEnvParser = NewSecretEnvParser()
```

Comment:
According to the implementation of the `rawEnvParser.Match` function in L47, I think the order of these parsers is very important. Please add some comments to clarify this.